.github/workflows/tests.yml

```diff
@@ -151,52 +210,41 @@ jobs:
           draft: true
           delete-branch: true
 
-      # return failure if any checks failed
-      # return failure if any checks were cancelled
-      # loop again if any checks are still in progress
-      # loop again if no checks have completed yet
-      # return failure if checks are unchanged since last loop and the checks are empty
-      # return success if checks are unchanged since last loop and at least once check is completed
-      # (else) loop again if checks have changed since last loop (avoid race conditions)
+      # wait for checks register, complete and fail if any required checks failed
       - name: Wait for all checks to finish
         if: steps.cpr.outputs.pull-request-number != ''
         env:
           flowzone_re: "^Flowzone"
+          filter_re: "^policy-?bot"
+          GH_TOKEN: ${{ steps.gh_token.outputs.token || secrets.FLOWZONE_TOKEN }}
         run: |
-          while true
-          do
-            sleep 30
-
-            prev_checks="${checks}"
-
-            checks="$(gh pr status --json statusCheckRollup,number --jq '[.createdBy[] |
-              select(.number == ${{ steps.cpr.outputs.pull-request-number}}) | .statusCheckRollup[] |
-              select(.__typename == "CheckRun") |
-              select(.name | startswith("Flowzone")) |
-              {conclusion,status,name}]' || true)"
-
-            echo "${checks}" | jq
-
-            jq -r '.[].conclusion' <<< "${checks}" | grep -q '^FAILURE$' && exit 1
-            jq -r '.[].conclusion' <<< "${checks}" | grep -q '^CANCELLED$' && exit 1
-
-            jq -r '.[].status' <<< "${checks}" | grep -q '^IN_PROGRESS$' && continue
-            jq -r '.[].status' <<< "${checks}" | grep -q '^QUEUED$' && continue
-            jq -r '.[].status' <<< "${checks}" | grep -q '^PENDING$' && continue
-
-            if [ "${prev_checks}" == "${checks}" ]
-            then
-              [ $(jq 'length' <<< "${checks}") -gt 0 ] || exit 1
-              jq -r '.[].status' <<< "${checks}" | grep -q '^COMPLETED$' && exit 0
-            fi
+          while ! gh pr checks --required \
+            --repo ${{ matrix.repository }} \
+            ${{ steps.cpr.outputs.pull-request-number}} | grep -qE "${flowzone_re}"; do
+              echo '::info::waiting for required checks to register'
+              sleep $(((RANDOM % 5) + 5))
+          done
+          sleep $(((RANDOM % 15) + 15))
 
-            continue
+          while gh pr checks --required \
+            --repo ${{ matrix.repository }} \
+            ${{ steps.cpr.outputs.pull-request-number}} \
+            | grep -vE "${filter_re}" \
+            | grep -E '\s+pending\s+'; do
```

Comment:
I think ~~if~~ when we change the name of a required job we will get stuck here.

Because you are filtering on required, and looping if they are still pending. These PRs don't update the required checks to match the new names, so a renamed/refactored job will be missing and pending forever.

Reply:
Maybe if that's the case we can force merge the Flowzone PR, but it will come up at some point.

Reply:
.. and we also have these regexes in `parse_prepare_protection_rules`. Probably can leave it as is for now and if it ever comes up, we can look at parametrising the names?

Reply:
SGTM, maybe open a GH issue to investigate later